protocols/protocol/cloudstate/crdt.proto

```diff
@@ -51,20 +56,41 @@ option go_package = "cloudstate/protocol";
 // Special care must be taken when working with maps and sets. The keys/values are google.protobuf.Any, which encodes
 // the value as binary protobuf, however, serialized protobufs are not stable, two semantically equal objects could
 // encode to different bytes. It is the responsibility of the user function to ensure that stable encodings are used.
+//
+//---
+//
+//`Specification`:
+//
+//  - An implementation of `Crdt` is only valid iff:
+//  - `handle`
+//    - TODO: After invoking handle, the first message sent will always be a CrdtInit message, containing the entity ID, and,
+//      if it exists or is available, the current state of the entity. After that, one or more commands may be sent,
+//      as well as deltas as they arrive, and the entire state if either the entity is created, or the proxy wishes the
+//      user function to replace its entire state.
+//    - TODO: The user function must respond with one reply per command in. They do not necessarily have to be sent in the same
+//      order that the commands were sent, the command ID is used to correlate commands to replies.
```

Comment:
How would the order of replies can go out of order? This would imply that command replies can stream back concurrently and therefore possibly out of order to their arrivals sequence, right?

Reply:
@jroper wdyt?

Reply:
or this is meant to be _the "command ID is used to correlate commands to replies **and messages**_", the ones generated by subscription streaming messages, no? A Reply is here then both a CrdtReply or a CrdtStreamedMessage. And the order is relevant because these, and implementation specific here solely, depends how subscription callbacks where registered.

The only other reason to mention that could be, that commands theoretically get handled concurrently, or, they're allowed to.

@viktorklang @jroper … would you like to give me a hint here? I find nothing in the implementation that would support the latter paragraph. Thanks.

Reply:
@marcellanz Hmmm, good question. I would presume that @jroper would be more in tune with this than I am at the moment. What does the Proxy currently do?

Reply:
@viktorklang thanks, good hint. so far I intentionally did not read what the proxy does but guided the implementation on the spec, documentation and scala/node support for it, and therefore forgot I could :).
I'm still interested in @jroper's intent on it.

Reply:
The protocol supports commands being handled concurrently, the implementation of the proxy only issues one command at a time to crdts and event sourced entities, though i think may issue concurrent commands to stateless "entities".

Reply:
But yes, streamed messages get correlated to the original command by the command id.

Reply:
I agree that stateless entities can, and should, support parallel calls. Unless the developers assumes sequential processing and mistakenly considers this to be a premise in their function logic. Therefore, good documentation can guide developers to keep parallelism in mind.

Reply:
>The protocol supports commands being handled concurrently

I found _state changes_ can get handled concurrently and their resulting streamed messages therefore be sent to the proxy out of order. This might be useful if an entity has many streamed commands running, where their user functions take enough time to handle the change related to the time it takes to send their resulting streamed messages back to the proxy.

I'm not sure how much concurrent handling of state changes can help since sending actions and effects have to be serialised when sent down the grpc stream and I'm not sure if there are use cases worth it.

> Spec: They do not necessarily have to be sent in the same order that the commands were sent

I do realise re-reading this sentence not even has to imply concurrent processing of state changes, but the fact that the order of handling of state changes (EntityRunner#notifySubscribers in the Scala implementation) related to the command sequence does not have to be in the same order.